### PR TITLE
linux: make NumberUtils.cpp build

### PR DESCRIPTION
**Pull request by xanlpz**
- Provide 32 and 64bit implementations for NumberUtilities::MulLu
- Fallback gracefully in NumberUtilities::TryToInt64

OK, I feel like I'm constantly providing implementations for this large int multiplication thing, but here's another one.

### CODE REVIEW

File: lib/common/common/NumberUtilities.cpp [hunk 1]
```diff
@@ -76,8 +76,14 @@ namespace Js
         return (ulong)llu;
 #endif //!I386_ASM
 
-#else
-#error Neither _WIN32, nor _WIN64 is defined
+#elif defined(_M_X64) //!WINDOWS, 64bit
+        __uint128_t result = (__uint128_t)lu1 * (__uint128_t)lu2;
```
Comment (caitp) [inline lib/common/common/NumberUtilities.cpp]: `pal/inc/rt/intsafe.h` implements `UInt32x32To64`, is there any reason we don't use the one defined there?

**Comment by xanlpz** [comment]
Hrm, you mean using ClrSafeInt in safemath.h? I don't seem to have pal/inc/rt/intsafe.h in my tree.

**Comment by caitp** [comment]
hmm, you're right, it's not in the tree. Maybe it's from coreclr or something

**Comment by jianchun** [comment]
```https://github.com/Microsoft/ChakraCore/blob/linux/pal/inc/rt/intsafe.h``` is now available (I had it in a PR and checked in today.)

**Comment by xanlpz** [comment]
OK, pushed a new version using intsafe.h The bots are failing because apparently that is not available in git master at this point.

**Comment by xanlpz** [comment]
<intsafe.h> for Windows and the PAL include otherwise?

**Comment by obastemur** [comment]
For both it should be just fine.

**Comment by xanlpz** [comment]
OK I see, so the includes in rt/ actually have the same name as in Windows so you can just do that. Makes sense now, I was not getting the whole thing.

**Comment by xanlpz** [comment]
Pushed a commit building SmartFPUControl. As I say in the commit message there does not seem to be Linux APIs for this (google suggests just writing assembler for each architecture you target), and to be honest I do not even quite get why this is being used in the parser so porting it blindly at this point seems adventurous.

**Comment by jianchun** [comment]
For SmartFPUControl, I also looked at it the other day and opened #260. It is not used correctly. So we might need to implement it in future. I agree with your current change. Could you please update that commit description with this info? One nitpick: just for consistency, maybe replace ```!Windows``` with ```!_Win32```?

**Pull request merged**